arrow-libs/core/arrow-raise-contextual/src/commonMain/kotlin/arrow/core/raise/contextual/Raise.kt

```diff
@@ -0,0 +1,125 @@
+@file:OptIn(ExperimentalTypeInference::class, ExperimentalContracts::class)
+
+package arrow.core.raise.contextual
+
+import arrow.core.Either
+import arrow.core.NonEmptyList
+import arrow.core.NonEmptySet
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.raise.*
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+import kotlin.experimental.ExperimentalTypeInference
+import kotlin.jvm.JvmName
+import arrow.core.raise.Raise as CoreRaise
+import arrow.core.raise.SingletonRaise as CoreSingletonRaise
+import arrow.core.raise.ResultRaise as CoreResultRaise
+
+public typealias Raise<Error> = CoreRaise<Error>
+public typealias SingletonRaise<Error> = CoreSingletonRaise<Error>
+public typealias ResultRaise = CoreResultRaise
+
+context(raise: Raise<Error>) @RaiseDSL public fun <Error> raise(e: Error): Nothing =
+  raise.raise(e)
+
+context(raise: Raise<Error>) @RaiseDSL public inline fun <Error> ensure(condition: Boolean, otherwise: () -> Error) {
+  contract { returns() implies condition }
+  raise.ensure(condition, otherwise)
+}
+
+context(raise: Raise<Error>) @RaiseDSL public inline fun <Error, B : Any> ensureNotNull(value: B?, otherwise: () -> Error): B {
+  contract { returns() implies (value != null) }
+  return raise.ensureNotNull(value, otherwise)
+}
+
+context(raise: Raise<Error>) @RaiseDSL public inline fun <Error, OtherError, A> withError(
+  transform: (OtherError) -> Error,
+  @BuilderInference block: CoreRaise<OtherError>.() -> A
```

Comment:
```suggestion
  @BuilderInference block: Raise<OtherError>.() -> A
```
better to use the alias for consistency?